rfd/0048-environment-headers.md

```diff
@@ -0,0 +1,34 @@
+---
```

Comment:
Fixed!

Can do. Would you like me to update the implementation or will you?

Reply:
Actually, on further consideration, we have a few issues with this approach.

First is that the secret is dynamic. It rotates frequently and on an unpredictable basis, so in practice this file will need to be updated immediately before every `tsh` invocation.

Second—and more important—is that the secret is proxy-specific. It *must not* be sent to the wrong server, and should only be used when connecting to a specific proxy. We could add the secret immediately before `tsh` and remove it immediately after, but that leaves a window where another `tsh` invocation to a different location sends along a critical secret. And of course, cleanup may never even run for a variety of reasons.

Both of these combined lead me to the conclusion that a configuration file is not the best means of specifying this behavior.

Reply:
Do you have a wrapper that sets `TELEPORT_WEBCLIENT_HEADERS` based off the proxy you are trying to access? Because otherwise you'd still have the same problem because `TELEPORT_WEBCLIENT_HEADERS` will be applied to all `tsh` invocations?

Reply:
That is correct. We have a wrapper that sets this for the `tsh` invocation.

Reply:
@klizhentas @xinding33 Based off the requirements above (needing to define headers per-cluster), what do you think about the following?

```
$ cat ~/.tsh/headers
us-west-1.example.com:
  foo: bar
  baz: qux
us-west-2.example.com:
  foo: bar
  baz: qux
  ```

Reply:
That's a small improvement, but we will need to update this file with every `tsh` invocation anyway.

What's the underlying draw here for this to be configurable with a file on-disk and not through environment variables? I'm trying to understand the advantages you're seeing in this approach so we can come up with something that works best.

Reply:
https://github.com/gravitational/teleport/issues/9838

Reply:
We'd prefer not having to update a YAML file every `tsh` invocation, but if that's the approach you prefer to use then we can make it work.